### ISSUE TITLE
[bug] JSON imports broken

### ISSUE THREAD

**Issue by niedzielski** [lucacasonato/esbuild_deno_loader#46]
Thanks for the cool tool! This is a small bug report.

When I try to import any JSON such as the following:

```ts
import gamepadMapJSON from './src/input/gamepad/gamepad-map.json' assert {
  type: 'json',
}
```

I get an error:

```
✘ [ERROR] The file "src/input/gamepad/gamepad-map.json" was loaded with the "js" loader

    mod.ts:48:27:
      48 │ import gamepadMapJSON from './src/input/gamepad/gamepad-map.json' assert {
         ╵                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  This import assertion requires the loader to be "json" instead:

    mod.ts:49:2:
      49 │   type: 'json',
         ╵   ~~~~~~~~~~~~
```

If I omit the assertion, Deno complains `The module is a JSON module and not being imported with an import assertion. Consider adding `assert { type: "json" }` to the import statement.deno(no-assert-type)` (and I can't seem to ignore that).

I think I should just be able to add a loader to esbuild:

```
  loader: {
    '.json': 'json',
  },
```

But that seemed to have no effect. I tried to track it down and I think esbuild_deno_loader is maybe somehow [intercepting the JSON import](https://github.com/lucacasonato/esbuild_deno_loader/blob/0ee6541/src/portable_loader.ts#L257)? I'm working around this with the following JSON plugin:

```ts
    {
      name: 'json',
      setup: (build) =>
        build.onLoad({ filter: /\.json$/ }, () => ({ loader: 'json' })),
    },
```

In case it's helpful, the whole script looks like this:

```ts
const options: esbuild.BuildOptions = {
  sourcemap: 'linked',
  bundle: true,
  entryPoints: [input],
  format: 'esm',
  logLevel: `warning`,
  outfile: output,
  treeShaking: true,
  minify: true,
  plugins: [
    // Hack around JSON loader overrides in esbuild_deno_loader that flag type
    // assertions.
    {
      name: 'json',
      setup: (build) =>
        build.onLoad({ filter: /\.json$/ }, () => ({ loader: 'json' })),
    },
    denoPlugin({
      ...(importMap != null && { importMapURL: new URL(importMap) }),
    }) as unknown as esbuild.Plugin,
  ],
}

await esbuild.build(options)
esbuild.stop()
```

**Comment by lucacasonato** [comment]
Nice, esbuild fixed the behaviour of the `json` loader to only return JSON in the default export recently: https://github.com/evanw/esbuild/commit/b1e737b2e8bd231cd919337ada06c4ca994bd350. I can remove my workaround that is impacting you :)

**Comment by MrMarble** [comment]
Still broken on latest version (0.7) with esbuild 0.17.18, using the @niedzielski workaround works

**Comment by lucacasonato** [comment]
Fixed by #69
